### PR TITLE
[PTX-22914] Enhance UpgradeLongevity

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR implements TriggerUpgradeCluster to enable cluster upgrade capability for the UpgradeLongevity test suite. It ensures that IBM clusters use TriggerUpgradeVolumeDriverFromCatalog instead of TriggerUpgradeVolumeDriver and validates the licence summary after each upgrade. Px and Cluster upgrades are executed alternately, beginning with Px.

**Which issue(s) this PR fixes** (optional)
Closes #PTX-22914

**Special notes for your reviewer**:
~~I have rebased this branch onto `topic/krishna-px/PTX-22951` to test. I will provide an update once the test run has been completed.~~

Please review the following Jenkins builds:

1. SetupTeardown

https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-iks-catalog-longevity/4/

2. Px Upgrade 2.13.7 to 3.0.4

https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-iks-catalog-longevity/7/

3. Cluster Upgrade 1.27 to 1.28

https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-iks-catalog-longevity/8/

4. Px Upgrade 3.0.4 to 3.1.0.1, Cluster Upgrade 1.28 to 1.29, SetupTeardown

https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-iks-catalog-longevity/9/

Please note that we currently lack the Aetos Dashboard, Email Reporter, and Ginkgo Test reports for Longevity on IBM, hindering triage efforts. Thanks to @Balachandar-Pan for helping me understand IBM Cloud.

### CODE REVIEW

File: tests/common.go [hunk 1]
```diff
@@ -402,6 +403,124 @@ const (
 	pxctlCDListCmd = "pxctl cd list"
 )
 
+const (
+	IBMHelmRepoName   = "ibm-helm-portworx"
+	IBMHelmRepoURL    = "https://raw.githubusercontent.com/portworx/ibm-helm/master/repo/stable"
+	IBMHelmValuesFile = "/tmp/values.yaml"
```
Comment (lsrinivas-pure) [inline tests/common.go:409]: @kphalgun-px I thought we agreed moving all common constants to separate constants.go file

Comment (kphalgun-px) [reply tests/common.go:409]: Initially, I created a constants.go file but later realized removing all constants from common.go would be beyond the scope of this PR. I will address this in another PR.

**Review by Balachandar-Pan — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**